be/src/storage/tablet_updates.cpp

```diff
@@ -2492,6 +2493,17 @@ int64_t TabletUpdates::get_compaction_score() {
                 total_score += itr->second->compaction_score;
                 total_inputs += std::max(1UL, itr->second->num_segments);
                 total_deletes += itr->second->num_dels;
+                if (config::enable_pk_size_tiered_compaction_strategy) {
+                    int32_t level = _calc_compaction_level(itr->second.get());
+                    auto candidate_itr = candidates_by_level.find(level);
+                    if (candidate_itr == candidates_by_level.end()) {
+                        candidates_by_level[level] =
+                                std::make_pair(std::max(1UL, itr->second->num_segments), itr->second->num_dels);
+                    } else {
+                        candidate_itr->second.first += std::max(1UL, itr->second->num_segments);
```

Comment:
not consider overlapping?

Reply:
> not consider overlapping?

This records the number of segments, because segments are non-overlapping, and if the number of segments is greater than 1, it satisfies the compaction condition.

Reply:
> not consider overlapping?

This records the number of segments, because segments are non-overlapping, and if the number of segments is greater than 1, it satisfies the compaction condition.